sphinx/ext/autodoc.py

```diff
@@ -171,13 +171,15 @@ def __init__(self, names):
             #     set(['a', 'd'])
             self.base_packages.add(n.split('.')[0])
         self.mocked_modules = []  # type: List[str]
-        self.orig_meta_path = sys.meta_path
         # enable hook by adding itself to meta_path
         sys.meta_path = sys.meta_path + [self]
 
     def disable(self):
         # restore original meta_path to disable import hook
-        sys.meta_path = self.orig_meta_path
+        for idx, importer in enumerate(sys.meta_path):
+            if importer is self:
+                del sys.meta_path[idx]
+                break
```

Comment:
I appreciate your efforts to work carefully.
Since it is the cause of the problem that the original code does not work well when replacing with the backup of meta_path, I predict that replacing it with a new list does not work too.
So, could you try the code below?
```
sys.meta_path[:] = self.orig_meta_path
```